CentralityiOS/CentralityiOS/ViewControllers/AlertsModalViewController.m

```diff
@@ -0,0 +1,174 @@
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
+    return self.arrayOfPendingSharedTasks.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    TaskObject *task = self.arrayOfPendingSharedTasks[indexPath.row];
+    [task.owner fetchIfNeeded];
+    ReceiverCell *cell = [tableView dequeueReusableCellWithIdentifier:@"ReceiverCell" forIndexPath:indexPath];
+    cell.taskNameLabel.text = task.taskTitle;
+    cell.taskDescLabel.text = task.taskDesc;
+    cell.taskOwnerLabel.text = [NSString stringWithFormat:@"Owned by : %@", task.owner.username];
+    
+    NSString* allUsers = @"";
+    NSString* displayMessage = [[NSString alloc] init];
+    if (task.acceptedUsers.count > 0){
+        for (NSInteger index = 0; index < task.sharedOwners.count; index++){
+            allUsers = [allUsers stringByAppendingString:[task.acceptedUsers[index] fetchIfNeeded].username];
+            if (index < task.sharedOwners.count-1){
+                allUsers = [NSString stringWithFormat:@"%@, ",allUsers];
+            }
+        }
+        displayMessage = [NSString stringWithFormat:@"Accessible by : %@",allUsers];
+        cell.taskSharerLabel.hidden = FALSE;
+    }
+    else{
+        displayMessage = @"";
+        cell.taskSharerLabel.hidden = TRUE;
+    }
+        
+    cell.taskSharerLabel.text = displayMessage;
+    
+    return cell;
+}
+
+- (PFQuery*)queryAllPendingTasks{
+    PFQuery *receivedTasksQuery = [PFQuery queryWithClassName:kTaskClassName];
+    [receivedTasksQuery whereKey:kBySharedOwnerQueryKey equalTo:PFUser.currentUser];
+    [receivedTasksQuery whereKey:kByAcceptedUsersQueryKey notEqualTo:PFUser.currentUser];
+    return receivedTasksQuery;
+}
+
+- (void)fetchReceivedTasks{
+    PFQuery *query = [self queryAllPendingTasks];
+    
+    [query findObjectsInBackgroundWithBlock:^(NSArray *users, NSError *error) {
+        if (users != nil) {
+            self.arrayOfPendingSharedTasks = [users mutableCopy];
+        } else {
+            NSLog(@"%@", error.localizedDescription);
+        }
+        [self.receiverTableView reloadData];
+    }];
+}
+
+- (UISwipeActionsConfiguration *)tableView:(UITableView *)tableView
+trailingSwipeActionsConfigurationForRowAtIndexPath:(NSIndexPath *)indexPath{
+    
+    TaskObject *task = self.arrayOfPendingSharedTasks[indexPath.row];
+    
+    UIContextualAction *declineAction = [UIContextualAction contextualActionWithStyle:UIContextualActionStyleDestructive title:@"Decline" handler:^(UIContextualAction * _Nonnull action, __kindof UIView * _Nonnull sourceView, void (^ _Nonnull completionHandler)(BOOL)) {
+        
+        PFQuery *query = [self queryAllPendingTasks];
+        
+        [query findObjectsInBackgroundWithBlock:^(NSArray *tasks, NSError *error) {
+            if (tasks != nil) {
+                for (int i = 0; i < task.sharedOwners.count; i++) {
```

Comment:
i think you could split out each of these `for` loops into a helper method that takes in an array of users & the current user's object id